CoreMemoryCache/NHibernate.Caches.CoreMemoryCache/CoreMemoryCache.cs

```diff
@@ -0,0 +1,305 @@
+		}
+
+		/// <summary>
+		/// Default constructor.
+		/// </summary>
+		public CoreMemoryCache()
+			: this("nhibernate", null)
+		{
+		}
+
+		/// <summary>
+		/// Constructor with no properties.
+		/// </summary>
+		/// <param name="region">The region of the cache.</param>
+		public CoreMemoryCache(string region)
+			: this(region, null)
+		{
+		}
+
+		/// <summary>
+		/// Full constructor.
+		/// </summary>
+		/// <param name="region">The region of the cache.</param>
+		/// <param name="properties">Cache configuration properties.</param>
+		/// <remarks>
+		/// There are three (3) configurable parameters:
+		/// <ul>
+		///		<li>expiration (or cache.default_expiration) = number of seconds to wait before expiring each item.</li>
+		///		<li>cache.use_sliding_expiration = a boolean, true for resetting a cached item expiration each time it is accessed.</li>
+		/// 	<li>regionPrefix = a string for prefixing the region name.</li>
+		/// </ul>
+		/// All parameters are optional. The defaults are an expiration of 300 seconds, no sliding expiration and no prefix.
+		/// </remarks>
+		/// <exception cref="ArgumentException">The "expiration" property could not be parsed.</exception>
+		public CoreMemoryCache(string region, IDictionary<string, string> properties)
+		{
+			RegionName = region;
+			Configure(properties);
+		}
+
+		/// <inheritdoc />
+		public string RegionName { get; }
+
+		/// <summary>
+		/// The expiration delay applied to cached items.
+		/// </summary>
+		public TimeSpan Expiration { get; private set; }
+
+		/// <summary>
+		/// Should the expiration delay be sliding?
+		/// </summary>
+		/// <value><see langword="true" /> for resetting a cached item expiration each time it is accessed.</value>
+		public bool UseSlidingExpiration { get; private set; }
+
+		private void Configure(IDictionary<string, string> props)
+		{
+			var regionPrefix = DefaultRegionPrefix;
+			if (props == null)
+			{
+				Log.Warn("Configuring cache with default values");
+				Expiration = DefaultExpiration;
+				UseSlidingExpiration = _defaultUseSlidingExpiration;
+			}
+			else
+			{
+				Expiration = GetExpiration(props);
+				UseSlidingExpiration = GetUseSlidingExpiration(props);
+				regionPrefix = GetRegionPrefix(props);
+			}
+
+			_fullRegion = regionPrefix + RegionName;
+		}
+
+		private static string GetRegionPrefix(IDictionary<string, string> props)
+		{
+			if (props.TryGetValue("regionPrefix", out var result))
+			{
+				Log.Debug("new regionPrefix: {0}", result);
+			}
+			else
+			{
+				result = DefaultRegionPrefix;
+				Log.Debug("no regionPrefix value given, using defaults");
+			}
+
+			return result;
+		}
+
+		private static TimeSpan GetExpiration(IDictionary<string, string> props)
+		{
+			var result = DefaultExpiration;
+			if (!props.TryGetValue("expiration", out var expirationString))
+			{
+				props.TryGetValue(Cfg.Environment.CacheDefaultExpiration, out expirationString);
+			}
+
+			if (expirationString != null)
+			{
+				try
```

Comment:
Please change try-catch-int.Parse to int.TryParse.

Reply:
Old logic copy pasted from other caches, I have not even read it again indeed.